client.go

```diff
@@ -143,6 +143,91 @@ func (c *client) createSP(
 	return &result, password, err
 }
 
+// addAppPassword adds a new password to an App's credentials list.
+func (c *client) addAppPassword(ctx context.Context, appObjID string, duration time.Duration) (string, string, error) {
+	keyID, err := uuid.GenerateUUID()
+	if err != nil {
+		return "", "", err
+	}
+
+	// Key IDs are not secret, and they're a convenient way for an operator to identify Vault-generated
+	// passwords. These must be UUIDs, so the three leading bytes will be used as an indicator.
+	keyID = "ffffff" + keyID[6:]
```

Comment:
why not `vault1` or something like that?

Reply:
keyID must be a UUID string. `ffffff` was chosen since they'll sort to the bottom of the list in the Portal view. I'm assuming there might be lots of Vault keys but just a few manually created ones, which should be more prominent.